### ISSUE TITLE
How to keep an item at a fixed position in a Sortable list?

### ISSUE THREAD

**Issue by AaronHarris** [jasonslyvia/react-anything-sortable#44]
I have a Sortable list where I want between each SortableItem to be a fixed item (it happens to be an arrow glyphicon, but that's besides the point).

```
class SortableExample extends React.Component {
  _handleSort(data) {
    console.log(data);
  }

  render() {
    return (
      <Sortable onSort={this._handleSort.bind(this)}>
        <SortableItem className="item-1" sortData="react" key={1}>
          React
        </SortableItem>
        <FixedItem style={{float: 'left'}} />
        <SortableItem className="item-2" sortData="angular" key={2}>
          Angular
        </SortableItem>
        <FixedItem style={{float: 'left'}} />
        <SortableItem className="item-3" sortData="backbone" key={3}>
          Backbone
        </SortableItem>
      </Sortable>
    );
  }
}
```

In my attempt, if I try to sort the data, the FixedItems get pushed to the outside (FixedItems do not have the @sortable decorator/mixin), and the handlesort method receives `[undefined, "angular", "backbone","react",undefined]`. However If I go into browser devtools and move the FixedItem DOM nodes back into position, they remain fixed where they're supposed to.

Would would be the simplest way to keep these items fixed while keeping them at the same sibling level as the SortableItems?

**Comment by jasonslyvia** [comment]
It's currently unsupported but I think it's achievable.

**Comment by AaronHarris** [comment]
Hmm, I can think of a few different instances where you'd want to specify the "allowed" positions that an element should be able to take (eg. only every even position, or every 5th position, or beginning or end, only next to some other type of element, etc). Which methods would one have to modify to support something like this?

I'm assuming the right methods would be `swapArrayItemPosition` (to control how / which items were swapped) and either `calculateNewIndex` or `getIndexByOffset`? I'm think the easiest way to support (just my use case - every other item) would be to mod 2 any newindex that would be assigned to the placeholder (I already have it so that when the user drops the item, it re-renders with the proper elements in the right spot).

**Comment by jasonslyvia** [comment]
If we were to add this new feature, I'd like it to be adoptable by anyone other than only your case. 

The solution I'm thinking of is providing an option like `sortTarget`, and only children's className match sortTarget will be sortable, anything else remains untouched.

I'll let you know if it works out, also a PR is warmly welcomed!

**Comment by jasonslyvia** [comment]
I kind of got it working and here's a brief demo, is it what you expected?

![may-07-2016 20-17-59](https://cloud.githubusercontent.com/assets/1336484/15092068/e1746b8e-1490-11e6-81ec-6f1275b59a2e.gif)

**Comment by AaronHarris** [comment]
Yes that's exactly what I was looking for! Much appreciated!

**Comment by jasonslyvia** [comment]
Cool, I'll add according tests and release a new version.

**Comment by jasonslyvia** [comment]
released `1.4.5`
